src/syntaxtree.c

```diff
@@ -236,15 +366,66 @@ static Obj SyntaxTreeRecExpr(Obj result, Expr expr)
     return result;
 }
 
-static Obj SyntaxTreeFloatLazy(Obj result, Expr expr)
+static Expr SyntaxTreeCodeRecExpr(Obj node)
 {
-    Obj string;
+    UInt1 tnum = GetTypeTNum(node);
+    Obj   keyvalue = ElmRecST(tnum, node, "keyvalue");
+    UInt  len = LEN_LIST(keyvalue);
```

Comment:
```suggestion
    UInt  len = LEN_PLIST(keyvalue);
```

Reply:
I might be going the other way to @fingolfin, but I was a bit worried at the number of uses of the PLIST functions in this, which don't check they are PLISTs. Using the LIST functions instead, while a little slower, would at least make sure GAP didn't crash/corrupt.

Reply:
I wouldn't mind using either the one or the other.